src/workerd/io/actor-cache.c++

```diff
@@ -1414,7 +1409,6 @@ kj::Own<ActorCache::Entry> ActorCache::findInCache(
   if (iter != ordered.end() && iter->get()->key == key) {
     // Found exact matching entry.
     Entry& entry = **iter;
-    KJ_ASSERT(entry.state != NOT_IN_CACHE, "entry should have been removed from map");
```

Comment:
I was a eyeing this assert but noticed it wasn't in the final diff so I'll ask here instead. Previously wondered the reasoning for checking `NOT_IN_CACHE` directly instead of `UNKNOWN`, but why delete this entirely?

Reply:
I could leave it in if you prefer! Basically, this particular assert doesn't have anything to do with this function, we have no reason why we cannot return the entry since we pulled it from the map just now. I'd also be willing to add `KJ_IASSERTS` to all the places we set `syncStatus` to make sure the previous state is correct.